apollo-router/src/plugins/csrf.rs

```diff
@@ -263,7 +263,10 @@ mod csrf_tests {
     #[tokio::test]
     async fn it_rejects_non_preflighted_headers_request() {
         let config = CSRFConfig::default();
-        let non_preflighted_request = RouterRequest::fake_builder().build().unwrap();
+        let non_preflighted_request = RouterRequest::fake_builder()
+            .header("content-type", "text/plain")
```

Comment:
does the header() method replace or append a header?

Raising this because the changelog mentions `When no Content-Type header is specified, this builder will now default to application/json which makes the request be accepted by CSRF protection.` so I'd suspect the header value to now be ["application/json", "text/plain"] which would be kinda surprising?